File: Swift-C++-IR-Comparison/test/arithmetic-operators/Substraction/Swift/OverflowSub.swift

@main
struct MyMain {

    static func main() {
        // This call would crash with a normal add. 
        // returns -9223372036854775808
        let a = subWithOverflow(x:Int.min, y:1)
    }

    // The & sign turns an subition to an substraction without overflow
    static func subWithOverflow(x: Int, y: Int) -> Int {
        let tmp = x &- y
        return tmp
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/OverflowSub.cpp
 
#include <iostream>
 
struct MyMain{
   static int subWithOverflow(int x) {
        int tmp = x & -1; 
        return tmp;
    }
   
    static void main(){
        // This call would crash with a normal add. 
        //returns -9223372036854775808 
        int a = subWithOverflow(x: INT_MIN); 
    }

};
  
int main(){
    MyMain::main();
}